src/interface/pipe.rs

```diff
@@ -36,6 +67,17 @@ pub struct EmulatedPipe {
 }
 
 impl EmulatedPipe {
+    /// # Description
+    /// Creates an in-memory pipe object
+    ///
+    /// # Arguments
+    ///
+    /// * `size` - Size of the iPC construct (either pipe or Unix socket)
+    ///
+    /// # Returns
+    ///
+    /// EmulatedPipe object
+    ///
```

Comment:
Feels like you either need the one above or this, not both.

Can you end up with ENOMEM or similar if you pass a huge value as the size?

Reply:
Cool. As for ENOMEM, thats not a pipe specific error code, and it only takes up to usize. This constructor is also only called with our constants for pipe and domain socket buf sizes and we don't alllow for adjusting that.